src/libraries/Common/src/System/Security/Cryptography/ECOpenSsl.cs

```diff
@@ -134,5 +114,32 @@ private void FreeKey()
                 _key = null!;
             }
         }
+
+        internal static SafeEvpPKeyHandle GenerateECKey(int keySize)
+        {
+            SafeEvpPKeyHandle ret = GenerateECKeyCore(new ECOpenSsl(keySize), out int createdKeySize);
+            Debug.Assert(keySize == createdKeySize);
+            return ret;
+        }
+
+        internal static SafeEvpPKeyHandle GenerateECKey(ECCurve curve, out int keySize)
+        {
+            return  GenerateECKeyCore(new ECOpenSsl(curve), out keySize);
+        }
+
+        internal static SafeEvpPKeyHandle GenerateECKey(ECParameters parameters, out int keySize)
```

Comment:
`Generate` means "make up a private key", this is really `Import`.

```suggestion
        internal static SafeEvpPKeyHandle ImportECKey(ECParameters parameters, out int keySize)
```

Reply:
It's also a little weird that these new methods are in terms of EVP_PKEY, but their name is "ECKey".  Perhaps that'll feel less weird in the future when EC_KEY doesn't lift this high.

Reply:
Good point this is an import. For the ECKey it's just a coincidence - in my head it's `elliptic curve key` rather than `EC_KEY` and type is an implementation detail

Reply:
because I don't have a better ideas for names I'll leave `ECKey` in the names (`ImportECEvpPKey` sounds weird to me and everything else I came up with seemed too long)